src/openms/include/OpenMS/FORMAT/DATAACCESS/SwathFileConsumer.h

```diff
@@ -631,7 +634,7 @@ namespace OpenMS
         addMS1Map_();
       }
       ms1_consumer_->consumeSpectrum(s);
-      s.clear(false);
+      ms1_map_->addSpectrum(s); // keep spectrum in memory
```

Comment:
this is going to be problematic for some cases, e.g. it is not clear that a "dump to disk" consumer will keep all MS1 data in memory. The issue is that we have SWATH data > 200 GB where the MS1 data alone is over 50 GB and this will clearly lead to issues if this is done. We may have to introduce a flag to regulate this behavior, probably both for MS1 and MS2